### PR TITLE
Backport: Allow additionally parsing \r newlines in PEM files

### PR DESCRIPTION
**Pull request by complexspaces**
This PR backports #30 to the 1.x release branch with the intent of releasing `1.0.4` with the bugfix as requested [here](https://discord.com/channels/976380008299917365/1015156984007381033/1172219516697198703) on the project's Discord.

For the most part this is a clean cherry-pick and only required an update to the test to match the older API of `read_all`.

Additionally, this PR includes the release boilerplate for making a `1.0.4` release.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ poor and doing so doesn't address a meaningful threat model.
 [![Documentation](https://docs.rs/rustls-pemfile/badge.svg)](https://docs.rs/rustls-pemfile/)
 
 # Release history
+- 1.0.4 (TBD)
```
Comment (cpu) [inline README.md:12]: Can you set this to today's date? I can publish it afterwards.

**Review by cpu — APPROVED** [review]
Thanks!

File: README.md (hunk 1 above)
Comment (complexspaces) [reply README.md]: I've added the date in.

**Comment by cpu** [comment]
* Pushed [v1.0.4](https://github.com/rustls/pemfile/releases/tag/v%2F1.0.4) tag
* Published rustls-pemfile v1.0.4 at registry `crates-io`

**Pull request merged**